src/main/java/cat/udl/eps/softarch/demo/domain/Batch.java

```diff
@@ -16,10 +20,19 @@ public class Batch {
     @GeneratedValue(strategy = GenerationType.AUTO)
 
     private Long id;
-    private int amount = -1;
-    private Long latitude;
-    private Long longitude;
-    private ZonedDateTime date;
+    @NotNull
+    @Min(value = 1, message = "There must be one or more batches")
+    private Integer amount;
+
+    @NotNull
+    @Min(value = 0, message = "Weight must be positive")
+    private Float weight;
```

Comment:
Better to use Decimal instead of Float to avoid precision issues. Moreover, following https://www.baeldung.com/javax-validation, there is a more specific validation annotation if the desired outcome is to have a positive value, to use @Positive. Moreover, it excludes 0, which makes sense in this case as the batch of at least one seed should weight more than 0